### PR TITLE
Don't register some Bolt and RocksDB metrics when Postgres is enabled

### PR DESCRIPTION
**Pull request by connorgorman**
## Description

Declutter prometheus

Screenshot 2023-03-01 at 12 29 31 PM

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

CI will tell me if I tried to double register something

### CODE REVIEW

File: central/metrics/init.go [hunk 1]
```diff
@@ -11,10 +12,16 @@ var (
...
+		)
+	}
+
 	prometheus.MustRegister(
-		boltOperationHistogramVec,
-		rocksDBOperationHistogramVec,
-		dackboxOperationHistogramVec,
 		graphQLOperationHistogramVec,
 		graphQLQueryHistogramVec,
 		indexOperationHistogramVec,
```
Comment (dashrews78) [inline central/metrics/init.go:27]: are we using this one for Postgres?

**Review by dashrews78 — APPROVED** [review]
LGTM.  just the one question on whether we can axe `indexOperationHistogramVec` or not.

File: central/metrics/init.go (hunk 1 above)
Comment (connorgorman) [reply central/metrics/init.go:27]: Yeah, technically search queries go through it

Comment (connorgorman) [reply central/metrics/init.go:27]: Post RocksDB, let's move this to like a postgres_search_op_duration or something like that

Comment (dashrews78) [reply central/metrics/init.go:27]: sounds good.

**Pull request merged**